b1ddi/provider.go

```diff
@@ -40,6 +46,12 @@ func Provider() *schema.Provider {
 				Default:     "api/ddi/v1",
 				Description: "The base path is to indicate the API version and the product name.",
 			},
+			"provider_version": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     terraformVersion,
+				Description: "Terraform provider version, used in the API call headers",
+			},
```

Comment:
Is it terraform version, or a terraform provider version? There is a terraform provider version variable in main.go already. 

Look at https://github.com/infobloxopen/terraform-provider-b1ddi/blob/9c59571c4ce9f76d79de810a466cd53436d3f486/.goreleaser.yml#L17

Reply:
Ok perfect. Then I will remove these changes and use the values set in `.goreleaser.yml`

Reply:
>Why do you need this as an optional field?
   
Since these variables are only accessible from main, I created a optional variable to store this value to be used while setting headers in the API calls. We will be using these headers for API tracking 

>Is there any case, in which provider version should be changed?
I thought we already do this: See https://registry.terraform.io/providers/infobloxopen/b1ddi/latest (different versions of the provider available in the registry)

Reply:
I don't understand why we are exposing those internal fields in the schema.